modules/incentives/src/migration.rs

```diff
@@ -19,29 +19,91 @@
 use super::*;
 use crate::log;
 use frame_support::traits::OnRuntimeUpgrade;
+use sp_std::marker::PhantomData;
 
-/// Clear all DexSavingRewardRates storage
-pub struct ClearDexSavingRewardRates<T>(sp_std::marker::PhantomData<T>);
-impl<T: Config> OnRuntimeUpgrade for ClearDexSavingRewardRates<T> {
+type WithdrawnRewards = BTreeMap<CurrencyId, Balance>;
+
+/// Clear all PendingMultiRewards for specific Pool
+pub struct ClearPendingMultiRewards<T, GetPoolId>(PhantomData<T>, PhantomData<GetPoolId>);
+impl<T: Config, GetPoolId: Get<PoolId>> OnRuntimeUpgrade for ClearPendingMultiRewards<T, GetPoolId> {
 	fn on_runtime_upgrade() -> Weight {
+		let pool_id = GetPoolId::get();
 		log::info!(
 			target: "incentives",
-			"ClearDexSavingRewardRates::on_runtime_upgrade execute, will clear Storage DexSavingRewardRates",
+			"ClearPendingMultiRewards::on_runtime_upgrade execute, will clear Storage PendingMultiRewards for Pool {:?}",
+			pool_id,
 		);
 
-		// clear storage DexSavingRewardRates,
-		let _ = DexSavingRewardRates::<T>::clear(u32::max_value(), None);
+		// clear all PendingMultiRewards for specific pool
+		let _ = PendingMultiRewards::<T>::clear_prefix(pool_id, u32::max_value(), None);
 
 		0
 	}
 
 	#[cfg(feature = "try-runtime")]
 	fn post_upgrade() -> Result<(), &'static str> {
-		assert_eq!(DexSavingRewardRates::<T>::iter().count(), 0);
+		let pool_id = GetPoolId::get();
+		assert_eq!(PendingMultiRewards::<T>::iter_prefix(pool_id).count(), 0);
 
 		log::info!(
 			target: "incentives",
-			"ClearDexSavingRewardRates done!",
+			"ClearPendingMultiRewards for Pool {:?} done!",
+			pool_id,
+		);
+
+		Ok(())
+	}
+}
+
+/// Reset rewards record for storage rewards.SharesAndWithdrawnRewards and rewards.PoolInfos at
+/// specific PoolId
+pub struct ResetRewardsRecord<T, GetPoolId>(PhantomData<T>, PhantomData<GetPoolId>);
+impl<T: Config, GetPoolId: Get<PoolId>> OnRuntimeUpgrade for ResetRewardsRecord<T, GetPoolId> {
+	fn on_runtime_upgrade() -> Weight {
+		let pool_id = GetPoolId::get();
+		log::info!(
+			target: "rewards",
+			"ResetRewardsRecord::on_runtime_upgrade execute, will reset Storage SharesAndWithdrawnRewards and PoolInfos for Pool {:?}",
+			pool_id
+		);
+
+		let mut total_share: Balance = Default::default();
+
+		// reset SharesAndWithdrawnRewards
+		for (who, (share, _)) in orml_rewards::SharesAndWithdrawnRewards::<T>::iter_prefix(&pool_id) {
+			orml_rewards::SharesAndWithdrawnRewards::<T>::mutate(&pool_id, &who, |(_, withdrawn_rewards)| {
+				*withdrawn_rewards = WithdrawnRewards::new();
+			});
+
+			total_share = total_share.saturating_add(share);
```

Comment:
The WithdrawnRewards is BTreeMap, e.g.
```
{
        {token:ACA}: 2,047,606,129,239,216
        {token:AUSD}: 62,077,309,016,549,947,911
        {foreignAsset:4}: 63,305,740,673,477
}
```

But the total_share is Balance, I'm not sure the result is correct.

Reply:
https://github.com/open-web3-stack/open-runtime-module-library/blob/master/rewards/src/lib.rs#L80

Reply:
here is vaule definition for `SharesAndWithdrawnRewards` : 
https://github.com/open-web3-stack/open-runtime-module-library/blob/44fda4432b71f49ee59a650cf4775db895ab97af/rewards/src/lib.rs#L110

The previous one of the tuple is `Balance` type to record the staked LP token amount,  `WithdrawnRewards` records the withdrawn rewards for calculation unsettled rewards. 

And `rewards.PoolInfos` for aUSD-IBTC has been killed in before referendum. Therefore, in addition to clearing the unsettled rewards(`WithdrawnRewards` in `SharesAndWithdrawnRewards` ) and the rewards to be claimed(incentives.PendingMultiRewards), the total amount of staked LP token should be counted and reset in `rewards.PoolInfos`.